### DISCUSSION TITLE
Randomization & video-slider RT every second

### DISCUSSION THREAD

**Discussion by aschutz32** [jspsych/jsPsych#2939] [Q&A] [answered]
Hello - I am new to coding in general, so I would really appreciate some pointers. 

I need to set up an online experiement where participants watch a 15-min video and continuously respond to a behavioral question, with their responses recorded every second. I have four movies that I need to randomize participants to, as well as 9 variables I need to randomly assign as well. I have been trying to get familiar with jsPsych for a while now, but I am having some trouble figuring out how to randomly assign per movie and per measure, as well as how to update the video-slider-response plug-in to capture responses every second.

Any and all advice/help is welcome!

Thank you!

**Comment by ChristopheBossens** [comment, marked as answer]
Hi aschutz32,

The question of how you are going to solve the randomization question partly depends on how you are going to host the experiment? One (tedious) approach would be to create a link for each version of your experiment and distribute different links to different people. Another alternative is to use a platform such as cognition.run, where it is possible to work with inter experiment conditions . Specifically, it will inject a variable in your JavaScript code that you can use to select the proper video/variables. Perhaps this is an option that also exists on other platforms but I am not aware of those. Of course it is also possible to write your own back-end code to take are of this if you have access to your own server.

From what I understand, you need a video-slider-response where the position of the slider is sampled every second, right? Do you think it would also work for your purposes if it simply records the data every time the slider position has been changed. Which version of jsPsych are you using? I think this is a fun little problem to work on, but it will require adjusting the code of the plugin itself.

**Reply by aschutz32** [reply]
Hello Christophe,

I am using jsPsych 7.3 to build the experiment. I was thinking I could use jsPsych.randomization.sampleWithoutReplacement to assign to movie and then to measure - would that be an option? I will be using either psiturk, cognition.run, or jatos to host the experiment online, so I am hoping to write the randomization code on the back-end, but I am not entirely sure how.

Also, yes I need the video-slider response to sample responses made every second. If it more simply just records data every time the slider position has changed, that could work as long as it is timestamped so I can match up to the timeline of the movie. 

Thank you so much for your input!! This is all very helpful.

**Reply by ChristopheBossens** [reply]
I think I have managed to do it in such a way that it records the history of responses each time the value of the slider changes.

[plugin-video-slider-response.zip](https://github.com/jspsych/jsPsych/files/10437026/plugin-video-slider-response.zip)

If you download the jsPsych library locally, you have to overwrite the plugin-video-slider-response.js with the one from the zip file above. The following code displays a trial with a video and shows the data in de developer tools when the trial is finished:

```
const video_trial = {
	type: jsPsychVideoSliderResponse,
	stimulus: ['video.webm'],
	track_slider_history : true,
	on_finish: function(data) {
		console.log(data);
	}
}
```

If you're interested, I also wrote a [small blog post](https://www.copsy.be/blog/updating-the-video-slider-response-to-record-position-history) about the process of updating the plugin.

@jodeleeuw is this something that could be incorporated in the jsPsych library? Maybe other people might find it useful to have this feature?

**Reply by aschutz32** [reply]
Thank you so much! This will work perfectly!
